### DISCUSSION TITLE
nixvim integration

### DISCUSSION THREAD

**Discussion by marvin-tr-81** [nix-community/stylix#570] [Q&A]
Hi, 

I have configured home-manager as a module and I can use nixos-rebuild test to apply home-manager changes. 

I have also configured stylix successfully and have applied custom themes to kitty, bash etc. 

I have set `stylix.autoEnable = true` and my selected theme is applied to Firefox even though I do not set `stylix.targets.firefox = true`. 

But I can not apply themes to nixvim. I have set `stylix.targets.nixvim = true` even though it seems to be redundant. 

I can set nixvim colorscheme with `programs.nixvim.colorschemes.gruvbox.enable = true` in nixos config for example, but when I remove this line no colorscheme is applied. I have even tried `programs.nixvim.colorschemes.stylix.enable = true` with desperation but obviously this caused error messages.

It seems I have to configure nixvim  somehow but I couldn't find any documentation on this topic online.

Thanks in advance for your help.

**Comment by trueNAHO** [comment]
> I have also configured stylix successfully and have applied custom themes to kitty, bash etc.
>
> I have set `stylix.autoEnable = true`

This is the current default value:

https://github.com/danth/stylix/blob/eccb9f2d63f4582b1c1ffe97d806156147aeee5f/stylix/target.nix#L17-L32

> and my selected theme is applied to Firefox even though I do not set `stylix.targets.firefox = true`.

Do you mean that Stylix is overriding your theme that is not integrated with Stylix?

**Reply by marvin-tr-81** [reply]
> Do you mean that Stylix is overriding your theme that is not integrated with Stylix?

Stylix applies themes to Firefox even without any configurations. I guess this is the intended behavior. I just included this information to indicate that Stylix (I think) is configured properly.

However, the theme configured in Stylix is not applied to nixvim even when I explicitly enable nixvim as a target (I know this is not required). I can manually set nixvim colorscheme but not through Stylix.

Thanks for the help.

**Comment by marvin-tr-81** [comment]
I think I may have found the cause of the problem. I have set nixvim in nixos configuration, not home-manager. On the other hand, I use `stylix.homeManagerModules.stylix` so I believe that Stylix can not apply the themes to nixvim. Now I will try to set `stylix.nixosModules.stylix` in my `flake.nix` and change every configurations accordingly. I will update the thread with the results if I am successful.

**Reply by marvin-tr-81** [reply]
My theory turned out to be true. In summary, I used `stylix.nixosModules.stylix` and added stylix configurations into `configuration.nix`. Now, the themes set in stylix are also applied to nixvim.
